src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -174,6 +178,9 @@ protected void doXContent(XContentBuilder xContentBuilder, Params params) throws
         if (Objects.nonNull(minScore)) {
             xContentBuilder.field(MIN_SCORE_FIELD.getPreferredName(), minScore);
         }
+        if (Objects.nonNull(methodParameters)) {
+            MethodParametersParser.doXContent(xContentBuilder, methodParameters);
```

Comment:
is this class MethodParametersParser coming from k-NN?

Reply:
Yes

Reply:
This is good. Lets ensure that we add some more docs on the MethodParametersParser function that it is now going beyond the boundaries of k-NN plugin. I would hate to see if we change signature of these functions and then neural plugin starts to break.

Reply:
https://github.com/opensearch-project/k-NN/pull/1758/files